llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -2493,6 +2502,14 @@ SDValue AMDGPUTargetLowering::LowerFRINT(SDValue Op, SelectionDAG &DAG) const {
   return DAG.getNode(ISD::FROUNDEVEN, SDLoc(Op), VT, Arg);
 }
 
+SDValue AMDGPUTargetLowering::LowerLRINT(SDValue Op, SelectionDAG &DAG) const {
+  auto ResVT = Op.getValueType();
+  auto Arg = Op.getOperand(0u);
+  auto ArgVT = Arg.getValueType();
+  SDValue RoundNode = DAG.getNode(ISD::FROUNDEVEN, SDLoc(Op), ArgVT, Arg);
+  return DAG.getNode(ISD::FP_TO_SINT, SDLoc(Op), ResVT, RoundNode);
```

Comment:
I don't think this is a correct implementation. Is there a conformance test of some kind for this? This would belong in generic lowering code too

Reply:
Hmm, this is what musl does for the lround case (with an implicit cast, gross)
https://git.musl-libc.org/cgit/musl/tree/src/math/lround.c

But the lower lrint case is more complicated:
https://git.musl-libc.org/cgit/musl/tree/src/math/lrint.c

Can you split the PR into one for each intrinsic? (plus another for the verifier changes)

Reply:
The lrint complexity seems related to FE_INEXACT, which we don't care about for the non-strictfp case